test/language/expressions/assignment/destructuring/obj-prop-nested-array-yield-expr.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    When a `yield` token appears within the Initializer of a nested
+    destructuring assignment and within a generator function body, it should
+    behave as a YieldExpression.
+es6id: 12.14.5.4
```

Comment:
I'm curious, could you talk more about that? To me, this all flows naturally, but I've been looking at the grammar for too long to be a unbiased critic of rationality :P